charts/opentelemetry-operator/release/main.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"gopkg.in/yaml.v3"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+const (
+	operatorManifestURL = "https://github.com/open-telemetry/opentelemetry-operator/releases/latest/download/opentelemetry-operator.yaml"
+	collectorCRDPath = "../crds/crd-opentelemetrycollector.yaml"
+)
+
+// K8sObject describes the values a Kubernetes object typically has.
+type K8sObject struct {
+	ApiVersion string `yaml:"apiVersion"`
+	Kind string `yaml:"kind"`
+	Metadata map[interface{}]interface{} `yaml:"metadata"`
+	Spec map[interface{}]interface{} `yaml:"spec"`
+	Status map[interface{}]interface{} `yaml:"status"`
+}
+
+// The main keeps the OpenTelemetry Operator (OTEL Operator) Helm chart configurations up-to-date with the upstream automatically.
+// First, it scrapes the latest OTEL Operator manifest from the OTEL Operator github:
+// https://github.com/open-telemetry/opentelemetry-operator.
+// Then, it will update the OpenTelemetry Collector CRD YAML file, which is the only CRD at this point.
```

Comment:
Helm mentions that there isn't a consensus in the community about how to update a CRD and that it might cause data loss. Do you know more about that? Are we doing something to avoid this? As Helm decided that the best is to let admins handle CRD updates, shouldn't we do the same?

Reply:
Yes, CRD has limitations ([here](https://github.com/helm/community/blob/f9e06c16d89ccea1bea77c01a6a96ae3b309f823/architecture/crds.md#upgrading-and-modifying-crds) and [here](https://helm.sh/docs/topics/charts/#limitations-on-crds)). Lots of unwanted things could happen due to CRD's mutability, which means we need to be extremely cautious when we make changes to it. And `helm upgrade` command won't upgrade CRD.

But the code here just keeps the CRD manifest in the Helm chart the same with the [upstream Operator manifest](https://github.com/open-telemetry/opentelemetry-operator/releases/latest/download/opentelemetry-operator.yaml). I don't think it's the same thing with the risks we talked above. Please let me what you think. Thanks!

Reply:
Indeed, sorry for the confusion.